streaming_providers/torbox/client.py

```diff
@@ -53,8 +58,14 @@ def get_torrent_info(self, magnet_id):
                 return torrent
         return {}
 
-    def get_torrent_instant_availability(self):
-        return self.get_user_torrent_list().get("data", {})
+    def get_torrent_instant_availability(self, torrent_hashes: list[str]):
+        response = self._make_request(
+            "GET", "/torrents/checkcached", params={"hash": torrent_hashes, "format": "list"}
+        )
+        data = response.get("data", {})
+        if type(data) is dict and data.get("data", False) is False:
+            return []
+        return data
```

Comment:
```suggestion
        return response.get("data", [])
```

according to their doc, 
![image](https://github.com/mhdzumair/MediaFusion/assets/44891939/3f881bab-e3c7-45d0-a2b1-ce92667424a7)

The `data` will be a list. I'm not sure what the response if there is no cached torrents. Might be an empty list of `data` or without `data` property, so the suggested line is enough imo.

Reply:
Yeah, that's what they show in the doc, but what I observed in testing was :

* If there is at least one cached torrent, the data type of the field was a list
* If there are no cached torrents, the data type of the field was a dict, containing another field called data which had a type of boolean and said False. I would have expected just an empty list as the response, but it was quite different.

Reply:
<img width="428" alt="failure" src="https://github.com/mhdzumair/MediaFusion/assets/112108222/61792843-e8d6-4727-bccd-74e3aa9a1601">

Reply:
okay, in that case, let's only check for type dict. 

```suggestion
        data = response.get("data", {})
        if type(data) is dict:
            return []
        return data
```

Reply:
<img width="1045" alt="success" src="https://github.com/mhdzumair/MediaFusion/assets/112108222/de4a02e6-f2c2-493d-9528-d2e0f8df25e2">
<img width="680" alt="failure" src="https://github.com/mhdzumair/MediaFusion/assets/112108222/43803fd5-e730-4af7-a7aa-f70d5b87c261">

Reply:
I can do another flow, instead of format as list, I can send the request with format as object, that will return a dict in both scenarios and respond that way, what do you think? That at least makes the data type to be a dict in both scenarios.

Reply:
> Format you want the data in. Acceptable is either "object" or "list". List is the most performant option as it doesn't require modification of the list.

List is fine. At the end, we used that response list data in the list comprehensive

Reply:
<img width="686" alt="changes" src="https://github.com/mhdzumair/MediaFusion/assets/112108222/37c58818-c8fc-4a08-af93-56d7c79da0e4">

The object type looks a bit cleaner in the code ? Can avoid those weird checks etc?

Reply:
could you show me a response data?

Reply:
<img width="1105" alt="success" src="https://github.com/mhdzumair/MediaFusion/assets/112108222/94251893-e335-44b9-9034-20c32c1b83af">
<img width="669" alt="failure" src="https://github.com/mhdzumair/MediaFusion/assets/112108222/9a7d7301-3996-4a81-aaad-0c690c584bc1">

Reply:
perfect. its similar to the [RD](https://github.com/mhdzumair/MediaFusion/blob/main/streaming_providers/realdebrid/utils.py#L72C10-L72C10). push this changes.

Reply:
Cool, pushing now